fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/impl/resource/conditions/ResourceConditionsImpl.java

```diff
@@ -204,4 +209,41 @@ public static boolean tagsPopulatedMatch(JsonObject object, RegistryKey<? extend
 
 		return true;
 	}
+
+	public static ConditionJsonProvider featureEnabled(Identifier id, final FeatureFlag feature) {
+		final Identifier featureId = ((FeatureManagerAccessor) FeatureFlags.FEATURE_MANAGER).getFeatureFlags().entrySet()
+				.stream()
+				.filter(flag -> {
+					FeatureFlagAccessor f1 = (FeatureFlagAccessor) flag.getValue();
+					FeatureFlagAccessor f2 = (FeatureFlagAccessor) feature;
+					return f2.getMask() == f2.getMask() && f1.getUniverse().toString().equals(f2.getUniverse().toString());
+				})
+				.findFirst()
+				.map(Map.Entry::getKey)
+				.orElseThrow(() -> new IllegalArgumentException("Unknown feature passed"));
+
+		return new ConditionJsonProvider() {
+			@Override
+			public Identifier getConditionId() {
+				return id;
+			}
+
+			@Override
+			public void writeParameters(JsonObject object) {
+				object.addProperty("feature", feature.toString());
+			}
+		};
+	}
+
+	public static FeatureSet currentFeature = FeatureFlags.DEFAULT_ENABLED_FEATURES;
+
+	public static boolean featureEnabledMatch(JsonObject object) {
```

Comment:
Similarly, you can convert a list of identifiers to a `FeatureSet` as follows:

```java
			manager.featureSetOf(identifiers, id -> {
				throw new JsonParseException("Unknown feature flag: " + id);
			});
```

This will allow removing the other mixin.